crowbar_framework/app/models/api/upgrade.rb

```diff
@@ -143,9 +143,10 @@ def checks
                    reduce({}, :merge)
 
           if errors.any?
-            upgrade_status.end_step(false, errors)
+            ::Crowbar::UpgradeStatus.new.end_step(false, errors)
           else
-            upgrade_status.end_step
+            ::Crowbar::UpgradeStatus.new.save_upgrade_mode(ret[:best_method])
```

Comment:
this will not work when re-running prechecks after one successful run (it will return in line 125 above)